### PR TITLE
[DD4hep] DDShape access to a composite shape's right solid

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* GEM geometry builder requires access to both left and right solid of a composite shape

#### PR validation:

```DetectorDescription/DDCMS/test/DDSolid.cppunit.cc``` unit test 

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

File: DetectorDescription/DDCMS/src/DDFilteredView.cc [hunk 1]
```diff
@@ -24,6 +24,17 @@ dd4hep::Solid DDSolid::solidA() const {
   return solid_;
 }
 
+dd4hep::Solid DDSolid::solidB() const {
+  if (dd4hep::isA<dd4hep::SubtractionSolid>(solid_) || dd4hep::isA<dd4hep::UnionSolid>(solid_) ||
+      dd4hep::isA<dd4hep::IntersectionSolid>(solid_)) {
+    const TGeoCompositeShape* sh = (const TGeoCompositeShape*)solid_.ptr();
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/src/DDFilteredView.cc]:
```suggestion
    const TGeoCompositeShape* sh = static_cast<const TGeoCompositeShape*>(solid_.ptr());
```

Comment (cvuosalo) [reply DetectorDescription/DDCMS/src/DDFilteredView.cc]: Will a static cast work? Our coding rules recommend C++ style casts:
https://github.com/cms-sw/cms-sw.github.io/blob/master/cms_coding_rules.md#4----technical-coding-rules-1

Comment (ianna) [reply DetectorDescription/DDCMS/src/DDFilteredView.cc]: thanks, @cvuosalo - done

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

**Comment by slomeo** [comment]
I need this PR to be merged in the last IB, please can you make the "orp-pending" label to become green?

**Comment by silviodonato** [comment]
+1

**Pull request merged**